qui/tray/updates.py

```diff
@@ -126,44 +126,42 @@ def check_vms_needing_update(self):
         self.vms_needing_update.clear()
         self.obsolete_vms.clear()
         for vm in self.qapp.domains:
-            try:
-                updates_available = vm.features.get('updates-available', False)
-            except exc.QubesDaemonCommunicationError:
-                updates_available = False
-            if updates_available and \
-                    (getattr(vm, 'updateable', False) or vm.klass == 'AdminVM'):
+            updated: bool = qui.utils.check_update(vm)
+            supported: bool = qui.utils.check_support(vm)
+            if not updated:
                 self.vms_needing_update.add(vm)
-            try:
-                supported = qui.utils.check_support(vm)
-            except exc.QubesDaemonCommunicationError:
-                supported = True
             if not supported:
                 self.obsolete_vms.add(vm.name)
 
     def connect_events(self):
         self.dispatcher.add_handler('domain-feature-set:updates-available',
-                                    self.feature_set)
+                                    self.feature_change)
         self.dispatcher.add_handler('domain-feature-delete:updates-available',
-                                    self.feature_unset)
+                                    self.feature_change)
+        self.dispatcher.add_handler('domain-feature-set:skip-update',
+                                    self.feature_change)
+        self.dispatcher.add_handler('domain-feature-delete:skip-update',
+                                    self.feature_change)
         self.dispatcher.add_handler('domain-add', self.domain_added)
         self.dispatcher.add_handler('domain-delete', self.domain_removed)
         self.dispatcher.add_handler('domain-feature-set:os-eol',
-                                    self.feature_set)
+                                    self.feature_change)
 
-    def domain_added(self, _submitter, _event, vm, *_args, **_kwargs):
+    def domain_added(self, _submitter, _event, vmname, *_args, **_kwargs):
```

Comment:
This is wrong, "vm" is used as keyword argument, so you can't change its name. I'll fix it.

I should have put that through openQA earlier...

Reply:
Ops. Sorry for mistake. I wonder why it did not show up during the unit tests.  I guess there is no unit test for installing an outdated or obsolete template.